js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('fetchBidsAsks', undefined, params);
+        let method = undefined;
+        if (marketType === 'spot') {
+            method = 'spotPublicGetTickerBookTicker';
+            //
+            //     [
+            //       {
+            //         "symbol": "AEUSDT",
+            //         "bidPrice": "0.11001",
+            //         "bidQty": "115.59",
+            //         "askPrice": "0.11127",
+            //         "askQty": "215.48"
+            //       },
+            //     ]
+            //
+        } else if (marketType === 'swap') {
+            throw new BadRequest (this.id + ' fetchBidsAsks() is not available for ' + marketType + ' markets');
+        }
+        const response = await this[method] (query);
+        return this.parseTickers (response, symbols);
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        if (market['spot']) {
+            return await this.createSpotOrder (market, type, side, amount, price, params);
+        } else if (market['swap']) {
+            return await this.createSwapOrder (market, type, side, amount, price, params);
+        }
+    }
+
+    async createSpotOrder (market, type, side, amount, price = undefined, params = {}) {
+        const symbol = market['symbol'];
+        const orderSide = (side === 'buy') ? 'BUY' : 'SELL';
+        // TODO: needs postOnly handing here, possible with LIMIT_MAKER
+        const request = {
+            'symbol': market['id'],
+            'quantity': this.amountToPrecision (symbol, amount),
+            'side': orderSide,
+            'type': type.toUpperCase (),
+        };
+        if (price !== undefined) {
+            request['price'] = this.priceToPrecision (symbol, price);
+        }
+        const clientOrderId = this.safeString (params, 'clientOrderId');
+        if (clientOrderId !== undefined) {
+            request['newClientOrderId'] = clientOrderId;
+            params = this.omit (params, [ 'type', 'clientOrderId' ]);
+        }
+        const response = await this.spotPrivatePostOrder (this.extend (request, params));
+        //
+        // spot
+        //
+        //     {
+        //         "symbol": "BTCUSDT",
+        //         "orderId": "123738410679123456",
+        //         "orderListId": -1
+        //     }
+        //
+        return this.extend (this.parseOrder (response, market), {
+            'side': side,
+            'type': type,
+            'price': price, // TODO: should we do this?
+            'amount': amount,
+        });
+    }
+
+    async createSwapOrder (market, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        const symbol = market['symbol'];
+        const openType = this.safeInteger (params, 'openType');
+        if (openType === undefined) {
```

Comment:
isn't it a bit more intuitive to require a "margin" parameter and convert it to `openType` internally? Also if not provided can't we assume a default value?

Reply:
pushed. I'll resolve this one.